backend/app/core/storage.py

```
@@ -0,0 +1,99 @@
+import abc
+import logging
+from typing import Optional, Dict # Added Dict for mock_db
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)